packages/@aws-cdk/aws-servicecatalog/lib/product-stack.ts

```diff
@@ -60,8 +73,38 @@ export class ProductStack extends cdk.Stack {
       fileName: this.templateFile,
     }).httpUrl;
 
+    if (this._productVersionDetails.retentionStrategy == RetentionStrategy.RETAIN) {
+      this.writeTemplateToContext(cfn, templateHash);
+    }
+
     fs.writeFileSync(path.join(session.assembly.outdir, this.templateFile), cfn);
   }
+
+  /**
+   * Writes current template generated from Product Stack to a context directory.
+   *
+   * @internal
+   */
+  private writeTemplateToContext(cfn: string, templateHash: string) {
+    if (!fs.existsSync(PRODUCT_STACK_SNAPSHOT_DIRECTORY)) {
+      fs.mkdirSync(PRODUCT_STACK_SNAPSHOT_DIRECTORY);
+    }
+    const templateFileKey = `${this._productVersionDetails.productPathUniqueId}.${this._productVersionDetails.productStackId}.${this._productVersionDetails.productVersionName}.product.template.json`;
+    const templateFilePath = path.join(PRODUCT_STACK_SNAPSHOT_DIRECTORY, templateFileKey);
+    if (fs.existsSync(templateFilePath)) {
+      const previousTemplateHash = crypto.createHash('sha256').update(fs.readFileSync(templateFilePath)).digest('hex');
```

Comment:
Why do we do hashing in the first place? Can't we just compare the two template definitions? We have both of them, right?